### PR TITLE
Rancher registry standalone package

### PR DESCRIPTION
**Pull request by izaac**
Package to create Docker private registries for use in Rancher.

- This package creates a docker registry with auth or without auth using valid certificates.
- The Rancher package was updated to use `CATTLE_SYSTEM_DEFAULT_REGISTRY` during helm install if a `registry_fqdn` variable exists in corral.

### CODE REVIEW

**Comment by izaac** [comment]
https://github.com/rancher/qa-tasks/issues/51

**Review by sbstp — APPROVED** [review]
Looks OK to me, but I'm still pretty new to corral.

**Comment by izaac** [comment]
The latest update just removes unused package variables from manifest.yaml

File: templates/standalone_registry/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: registry
+description: |
+  An docker registry
```
Comment (jakefhyde) [inline templates/standalone_registry/manifest.yaml:3]:
```suggestion
  A docker registry
```

**Review by jakefhyde — CHANGES_REQUESTED** [review]
Left some comments (mostly just nits). Also some general comments left on https://github.com/rancherlabs/corral-packages/pull/8 are also applicable here

File: templates/rancher/overlay/opt/corral/rancher/install-rancher.sh [hunk 2]
```diff
@@ -7,16 +7,28 @@ helm repo update
...
   -n cattle-system \
   rancher rancher-latest/rancher
-
+else
+  helm upgrade \
+    --install \
+    --create-namespace \
+    --set hostname="${CORRAL_rancher_host}" \
+    --set "extraEnv[0].name=CATTLE_SYSTEM_DEFAULT_REGISTRY" \
+    --set "extraEnv[0].value=$CORRAL_registry_fqdn" \
```
[the remark this answers is not in the document]
Comment (igomez06) [reply templates/rancher/overlay/opt/corral/rancher/install-rancher.sh:26]: Formatting issues: https://github.com/rancherlabs/corral-packages/pull/8/files#r1161909034

File: templates/aws/registry_nodes/manifest.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    description: "What size ec2 instance to use for the nodes."
+  registry_fqdn:
+    type: string
+    readOnly: true
+    description: "Public DNS A record that directs to the registry node."
+  registry_ip:
+    type: string
+    readOnly: true
+    description: "Public IP address of the registry node"
+  registry_count:
```
Comment (igomez06) [inline templates/aws/registry_nodes/manifest.yaml:55]: Will be cases with more than one registry node?

Comment (izaac) [reply templates/aws/registry_nodes/manifest.yaml:55]: I think this is a remnant from when I begun following the `agent_nodes`' design.
But then stepped back on that after I realized that's not the same case for registries.

I'll remove it.

Comment (izaac) [reply templates/aws/registry_nodes/manifest.yaml]: Removed unused variable
No re-test needed.

**Comment by izaac** [comment]
Hi @jakefhyde could I ask for another review when you're available?

File: templates/rancher/overlay/opt/corral/rancher/install-cert-manager.sh [hunk 4]
```diff
@@ -13,4 +15,21 @@ helm upgrade \
...
+kubectl apply -f cert-manager/cert-manager-crd.yaml
+
+helm install cert-manager "./cert-manager-v${CORRAL_cert_manager_version}.tgz" \
+    --namespace cert-manager \
+    --set image.repository="${CORRAL_registry_fqdn}/quay.io/jetstack/cert-manager-controller" \
+    --set webhook.image.repository="${CORRAL_registry_fqdn}/quay.io/jetstack/cert-manager-webhook" \
+    --set cainjector.image.repository="${CORRAL_registry_fqdn}/quay.io/jetstack/cert-manager-cainjector" \
+    --set startupapicheck.image.repository="${CORRAL_registry_fqdn}/quay.io/jetstack/cert-manager-ctl"
+
+fi
```
Comment (jakefhyde) [inline templates/rancher/overlay/opt/corral/rancher/install-cert-manager.sh:33]: Formatting (indent)

File: templates/aws/registry_nodes/terraform/pools/main.tf [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  }
+}
+
+provider "random" {}
+provider "aws" {
+  access_key = var.aws_access_key
+  secret_key = var.aws_secret_key
+  region =  var.aws_region
+}
```
Comment (jakefhyde) [inline templates/aws/registry_nodes/terraform/pools/main.tf:16]:
```suggestion
provider "aws" {
  access_key = var.aws_access_key
  secret_key = var.aws_secret_key
  region     =  var.aws_region
}
```
Auto format file

**Review by jakefhyde — CHANGES_REQUESTED** [review]
Couple of nits

**Comment by izaac** [comment]
@jakefhyde I think I've addressed all the feedback from the last round of review. Could I have another pass?

**Review by igomez06 — APPROVED** [review]
Looks good at least on my end. Will wait for Jake's final approval.

**Review by jakefhyde — APPROVED** [review]

**Comment by igomez06** [comment]
@jakefhyde could you merge the PR, I'll rebase the changes into branch. Thank you!

**Comment by izaac** [comment]
This last change added a missing variable in the `registries.yaml` package manifest.

**Comment by izaac** [comment]
Are we good for merge here ?

**Comment by izaac** [comment]
Hi @jakefhyde please check the PR. It is approved.

**Pull request merged**